docs/application/native/guides/app-management/ui-app/dali-widget-app.md

```diff
@@ -0,0 +1,269 @@
+      public:
+        explicit SampleWidget( Internal::SampleWidget* sampleWidget );
+      };
+      }
+      ```
+
+   2. Create body class of custom widget by using `<sample-widget-impl.h>` header file in your application:
+
+      ```
+      #include <dali/public-api/adaptor-framework/widget-impl.h>
+
+      // HANDLE INCLUDES
+      #include "sample-widget.h"
+
+      namespace Demo
+      {
+      namespace Internal
+      {
+      class SampleWidget : public Dali::Internal::Adaptor::Widget
+      {
+      public:
+        SampleWidget();
+        ~SampleWidget();
+        static Demo::SampleWidget New();
+        virtual void OnCreate( const std::string& contentInfo, Dali::Window window );
+        virtual void OnTerminate( const std::string& contentInfo, Dali::Widget::Termination type );
+        virtual void OnPause();
+        virtual void OnResume();
+        virtual void OnResize( Dali::Window window );
+        virtual void OnUpdate( const std::string& contentInfo, int force );
+
+      protected:
+        // Undefined
+        SampleWidget(const SampleWidget&);
+        SampleWidget& operator=(SampleWidget&);
+      };
+      } // namespace Internal
+      } // namespace Dali
+      ```
+
+   3. Implement each lifecycle functions:
+
+      ```
+      #include "sample-widget-impl.h"
+      #include <dali-toolkit/dali-toolkit.h>
+      #include <dali/integration-api/debug.h>
+
+      using namespace Dali;
+      using namespace Dali::Toolkit;
+
+      namespace Demo
+      {
+      namespace Internal
+      {
+
+      Demo::CustomWidget SampleWidget::New()
+      {
+        IntrusivePtr<SampleWidget> impl = new SampleWidget();
+        Demo::SampleWidget handle = Demo::SampleWidget( impl.Get() );
+        return handle;
+      }
+
+      SampleWidget::SampleWidget(){}
+
+      SampleWidget::~SampleWidget(){}
+
+      void SampleWidget::OnCreate( const std::string& contentInfo, Dali::Window window )
+      {
+        // This widget will draw "Hello world!" text when widget is loaded.
+
+        TextLabel label = TextLabel::New("Hello world\n");
+        Dali::Stage::GetCurrent().Add(label);
+      }
+
+      void SampleWidget::OnTerminate( const std::string& contentInfo, Dali::Widget::Termination type )
+      {
+        DALI_LOG_ERROR("Widget instance terminated\n");
+      }
+
+      void SampleWidget::OnPause()
+      {
+        DALI_LOG_ERROR("Widget instance paused\n");
+      }
+
+      void SampleWidget::OnResume()
+      {
+        DALI_LOG_ERROR("Widget instance resumed\n");
+      }
+
+      void SampleWidget::OnResize( Dali::Window window )
+      {
+        DALI_LOG_ERROR("Widget instance resized\n");
+      }
+
+      void SampleWidget::OnUpdate( const std::string& contentInfo, int force )
+      {
+        DALI_LOG_ERROR("Widget instance updated\n");
+      }
+
+      } // Internal
+      } // Dali
+      ```
+
+   4. Include custom widget header and add Widget `CreatingWidgetFunction()` function for custom widget:
```

Comment:
Please rephase as the following:

For custom widget include custom widget header. Also, add `Widget CreatingWidgetFunction()` function: